src/Microsoft.Azure.WebJobs.Host/JobHost.cs

```diff
@@ -232,13 +232,14 @@ public async Task CallAsync(string name, IDictionary<string, object> arguments =
 
         private async Task CallAsyncCore(IFunctionDefinition function, object functionKey, IDictionary<string, object> arguments, CancellationToken cancellationToken)
         {
-            Validate(function, functionKey);
-
-            IFunctionInstance instance = CreateFunctionInstance(function, arguments);
-
-            IDelayedException exception = null;
+            if (function == null)
+            {
+                throw new InvalidOperationException($"'{functionKey}' can't be invoked from Azure WebJobs SDK. Is it missing Azure WebJobs SDK attributes?");
+            }
 
-            exception = await _context.Executor.TryExecuteAsync(instance, cancellationToken);
+            var instance = CreateFunctionInstance(function, arguments);
+            var callCancellationSource = CancellationTokenSource.CreateLinkedTokenSource(_shutdownTokenSource.Token, cancellationToken);
```

Comment:
Did you just happen to notice that we weren't looking at the shutdown token here? Or was this an issue you hit?

Reply:
I was investigating https://github.com/Azure/azure-webjobs-sdk/issues/2420 and other recent reports while doing this work. I'll add a test for this as well.